tests/conftest.py

```diff
@@ -17,15 +17,19 @@
 # Early diagnostic for failed imports
 import pybind11_tests
 
-if os.name != "nt":
-    # Full background: https://github.com/pybind/pybind11/issues/4105#issuecomment-1301004592
-    # In a nutshell: fork() after starting threads == flakiness in the form of deadlocks.
-    # It is actually a well-known pitfall, unfortunately without guard rails.
-    # "forkserver" is more performant than "spawn" (~9s vs ~13s for tests/test_gil_scoped.py,
-    # visit the issuecomment link above for details).
-    # Windows does not have fork() and the associated pitfall, therefore it is best left
-    # running with defaults.
-    multiprocessing.set_start_method("forkserver")
+
+@pytest.fixture(scope="session", autouse=True)
+def always_forkserver_on_unix():
+    if os.name != "nt":
```

Comment:
With this being moved to a function, I have a slight preference for

```
if os.name == "nt":
  return
```
Then the rest of the code unindented one level.
That's slightly more future proof, in case we want to add more conditions.
But up to you.